mapsforge-map-writer/src/main/java/org/mapsforge/map/writer/util/GeoUtils.java

```diff
@@ -77,35 +77,24 @@ public final class GeoUtils {
      *
      * @param way                 the way
      * @param geometry            the geometry
-     * @param tileCoordinate      the tile coordinate
-     * @param enlargementInMeters the bounding box buffer
+     * @param tileAsGeometry      the tile coordinate
      * @return the clipped geometry
      */
-    public static Geometry clipToTile(TDWay way, Geometry geometry, TileCoordinate tileCoordinate,
-                                      int enlargementInMeters) {
-        Geometry tileBBJTS = null;
-        Geometry ret = null;
-
-        // create tile bounding box
-        tileBBJTS = tileToJTSGeometry(tileCoordinate.getX(), tileCoordinate.getY(), tileCoordinate.getZoomlevel(),
-                enlargementInMeters);
+    public static Geometry clipToTile(TDWay way, Geometry geometry, Geometry tileAsGeometry) {
+        Geometry ret;
 
         // clip the geometry by intersection with the bounding box of the tile
         // may throw a TopologyException
         try {
-            if (!geometry.isValid()) {
-                // this should stop the problem of non-noded intersections that trigger an error when
-                // clipping
-                LOGGER.warning("invalid geometry prior to tile clipping, trying to repair " + way.getId());
-                geometry = JTSUtils.repairInvalidPolygon(geometry);
-                if (!geometry.isValid()) {
-                    LOGGER.warning("invalid geometry even after attempt to fix " + way.getId());
-                }
+            if (tileAsGeometry.getEnvelopeInternal().contains(geometry.getEnvelopeInternal())) {
+                ret = geometry;
+            } else {
+                ret = tileAsGeometry.intersection(geometry);
             }
-            ret = tileBBJTS.intersection(geometry);
```

Comment:
@menion The rendering problem seems to be caused here:
- Declare `ret` only as intersection like before and the rendering seems to work correctly.

- Also, why remove the repair section if geometry is not valid?
```java
if (!geometry.isValid()) {
    // this should stop the problem of non-noded intersections that trigger an error when
    // clipping
    LOGGER.warning("invalid geometry prior to tile clipping, trying to repair " + way.getId());
    geometry = JTSUtils.repairInvalidPolygon(geometry);
    if (!geometry.isValid()) {
        LOGGER.warning("invalid geometry even after attempt to fix " + way.getId());
    }
}
ret = tileAsGeometry.intersection(geometry);
```